core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -56,14 +56,14 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
   }
 
   /** Set a configuration variable. */
-  def set(key: String, value: String): SparkConf = {
+  def set(key: String, value: Any): SparkConf = {
```

Comment:
Ah, the problem is that this would break binary compatibility, I believe. I think this would have to be a new method. Are there cases in the code base where you can now replace a `set("foo", bar.toString)` with `set("foo", bar)`?